File: DataSummitSolution/DataSummitUI/src/app/efModels/gender.ts
import { AspNetUser } from "../account/models/asp-net-user";
import { Employee } from "./employee";

export class Gender {

	GenderId: number;
	Type: string;
	CreatedDate?: Date | string;
	UserId: string;
	AspNetUsers: AspNetUser[];
	Employees: Employee[];

	constructor(genderId?: number, type?: string, createdDate?: Date | string,
		userId?: string, aspNetUsers?: AspNetUser[], employees?: Employee[])
	{
		this.GenderId = genderId || 0;
		this.Type = type || "";
		this.CreatedDate = createdDate || new Date(Date.now()) || null;
		this.UserId = userId || "";
		this.AspNetUsers = aspNetUsers;
		this.Employees = employees;
	}
}

File: DataSummitSolution/DataSummitModels/DB/Gender.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace DataSummitModels.DB
{
    public partial class Gender
    {
        public Gender()
        {
            AspNetUsers = new HashSet<AspNetUser>();
            Employees = new HashSet<Employee>();
        }

        public byte GenderId { get; set; }
        public string Type { get; set; }
        public DateTime? CreatedDate { get; set; }
        public string UserId { get; set; }

        public virtual ICollection<AspNetUser> AspNetUsers { get; set; }
        public virtual ICollection<Employee> Employees { get; set; }
    }
}
